test/extensions.txt

```diff
@@ -672,31 +672,51 @@ Hi!
 
 [^unused]: This is unused.
 .
-<p>This is some text!<sup class="footnote-ref"><a href="#fn1" id="fnref1">1</a></sup>. Other text.<sup class="footnote-ref"><a href="#fn2" id="fnref2">2</a></sup>.</p>
-<p>Here's a thing<sup class="footnote-ref"><a href="#fn3" id="fnref3">3</a></sup>.</p>
-<p>And another thing<sup class="footnote-ref"><a href="#fn4" id="fnref4">4</a></sup>.</p>
+<p>This is some text!<sup class="footnote-ref"><a href="#fn:1" id="fnref:1">1</a></sup>. Other text.<sup class="footnote-ref"><a href="#fn:footnote" id="fnref:footnote">2</a></sup>.</p>
```

Comment:
I think this addition will still not get us quite to where we need to use this in github.  What we need is a guarantee that multiple comments on the same page do not unintentionally link to eachother's anchors.  It seems like with this approach that will still happen if comments reuse the same footnote labels (e.g. so far I've just been using labels like `[^1]`)

What do you think about one of the following?
 - adding some some random text as part of the anchor name
 - adding some pseudorandom text obtained by e.g. digesting the footnote target body

Reply:
Hrmmm, the random text is probably the most robust solution but on first thought I don't _love_ inserting random strings into labels.

The pseudrandom is more pleasant, but then you're still stuck with folks say copying the same input generating the same output.

Maybe the best way to tackle _that_ issue is to handle it in the pipeline phase, like how we handle header ids for tables of contents – that way we can scope it to the id of the rendered comment/issue/whatever and avoid randomness AND be guaranteed everything is properly scoped.

Reply:
I like this solution although I bet that @talum will not! 😂

Reply:
If this is handled in the pipeline phase can there be a public description of how it is done so tools that try to mimic GitHub's output can do so without guessing?

Header IDs are still not publicly described and apparently [not open](https://github.com/Flet/github-slugger/pull/25#issuecomment-505916780) to GitHub employees as well.

Reply:
I'm testing the pipeline out with these colon-separated hrefs. Something with this syntax is wonky...

Example:

```<a href="#fn:other-note">testing a link</a>```

<a href="#fn:other-note">testing a link</a>

^^ does not show up as a link

Reply:
Experimenting with this, it looks like our sanitizer strips links whose hrefs contain `:` since 

`<a href="#fn-other-note">this works</a>` <a href="#fn-other-note">this works</a>

so, meh, okay, let's use a dash instead!

Reply:
@UziTech thanks for bringing this up.  In general we intentionally don't document things like how we generate slugs and ids not because it's secret or anything but because these are implementation details; we don't want to unintentionally define an API contract that we need to adhere to.

However, I think we'd be happy to informally describe the algorithm we eventually choose for generating footnote anchor ids from the output of what is publicly specified here.  Please create an issue and mention me after the feature is released and I'll do my best to provide an answer.